### PR TITLE
Added script to extract privileges of users in pre-refresh and ran it in post-refesh

### PR DESCRIPTION
**Pull request by soodabhinav**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build was run locally and any changes were pushed
- [ ] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by Ranzo3** [comment]
@soodabhinav 2 comments:

1)  What you've written works, but it would be better to have the pre-refresh to name the spool file "dxh_set_grants_users.sql" and run that in post-refresh.  Having it be the same name makes it a little confusing.
2)  What you've written probably works fine for most permissions, but I like the approach here a little more:  https://oracle-base.com/dba/script?category=script_creation&file=user_ddl.sql .  It is more likely to get all the various permissions needed, and to work on more versions of oracle.  Can you use this approach?

**Comment by Ranzo3** [comment]
The naming is still not quite right.  (The main logic looks good).

pre-refresh should call a script called "extract"
"extract" should create a script called "set"
post-refresh should call a script called "set"

**Comment by soodabhinav** [comment]
> The naming is still not quite right. (The main logic looks good).
> 
> pre-refresh should call a script called "extract"
> "extract" should create a script called "set"
> post-refresh should call a script called "set"

Thanks for pointing out . It is fixed now . we will going to call dxh_extract_grants_user.sql script in pre-refresh and it will spool  dxh_set_grants_users.sql script in post-refresh script

File: DBTEMPLATE/post-refresh/sql/dxh_wrapper_script.sql [hunk 1]
```diff
@@ -4,4 +4,6 @@ REM @${SQLDIR}/dxh_reset_user_password.sql
 REM create user testuser1
 REM PROMPT *** executing dxh_custom_create_user.sql ***
 REM @${SQLDIR}/dxh_custom_create_user.sql
+REM PROMPT *** executing dxh_extract_grants_users.sql ***
```
Comment (Ranzo3) [inline DBTEMPLATE/post-refresh/sql/dxh_wrapper_script.sql:7]: These two lines should be dxh_set_grants_users.sql I think

Comment (soodabhinav) [reply DBTEMPLATE/post-refresh/sql/dxh_wrapper_script.sql]: Its fixed now

**Pull request merged**